components/global/mixer-adapter/cmd/mixer-adapter/main.go

```diff
@@ -21,32 +21,74 @@ package main
 import (
 	"crypto/tls"
 	"crypto/x509"
+	"database/sql"
+	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"net/http"
 	"os"
-	"strconv"
+	"sync"
+	"time"
 
+	"github.com/go-sql-driver/mysql"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 
 	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/adapter"
 	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/logging"
+	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/publisher"
+	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/signals"
+	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/store"
+	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/store/database"
+	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/store/file"
+	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/store/memory"
+	"github.com/cellery-io/mesh-observability/components/global/mixer-adapter/pkg/writer"
 )
 
 const (
-	defaultAdapterPort         int    = 38355
-	grpcAdapterCertificatePath string = "GRPC_ADAPTER_CERTIFICATE_PATH"
-	grpcAdapterPrivateKeyPath  string = "GRPC_ADAPTER_PRIVATE_KEY_PATH"
-	caCertificatePath          string = "CA_CERTIFICATE_PATH"
-	spServerUrlPath            string = "SP_SERVER_URL"
+	configFilePathEnv string = "CONFIG_FILE_PATH"
 )
 
-func main() {
-
-	port := defaultAdapterPort //Pre defined port for the adaptor. ToDo: Should get this as an environment variable
+type (
+	Config struct {
+		Mixer struct {
+			TLS struct {
+				Certificate   string `json:"certificate"`
+				PrivateKey    string `json:"privateKey"`
+				CaCertificate string `json:"caCertificate"`
+			} `json:"tls"`
+		} `json:"mixer"`
+		SpEndpoint struct {
+			URL                 string `json:"url"`
+			SendIntervalSeconds int    `json:"sendIntervalSeconds"`
+		} `json:"spEndpoint"`
+		Store struct {
+			FileStorage struct {
+				Path string `json:"path"`
+			} `json:"fileStorage"`
+			Database struct {
+				Host     string `json:"host"`
+				Port     int    `json:"port"`
+				Protocol string `json:"protocol"`
+				Username string `json:"username"`
+				Password string `json:"password"`
+				Name     string `json:"name"`
+			} `json:"database"`
+			InMemory struct {
+			} `json:"inMemory"`
+		} `json:"store"`
+		Advanced struct {
+			BufferSize             int `json:"bufferSize"`
+			BufferSizeFactor       int `json:"bufferSizeFactor"`
```

Comment:
What is the requirement of this parameter? Why is it not enough just to directly provide buffer size?